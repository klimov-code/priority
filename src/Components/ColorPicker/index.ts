export { ColorPicker } from './ColorPicker';
export { Container, Circle } from './styled';
