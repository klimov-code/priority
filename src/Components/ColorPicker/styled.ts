import styled, { css } from 'styled-components';

const Border = css`
  box-shadow: inset 0 0 0 2px #8cc7f2;
`;

const Container = styled.div`
  padding: 8px;
  box-sizing: border-box;
  border-radius: 5px;

  ${({ isActive }: { isActive: boolean }) => isActive && Border};
`;

const Circle = styled.div`
  width: 20px;
  height: 20px;
  background: #4a4a4a;
  border-radius: 50%;
  cursor: pointer;
`;

export { Container, Circle };
