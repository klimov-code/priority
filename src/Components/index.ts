export { Priority } from './Priority';
export { Editor } from './Editor';
export { Viewer } from './Viewer';
export { Header } from './Header';
