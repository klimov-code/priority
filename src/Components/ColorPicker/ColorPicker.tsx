import React, { useState, useEffect } from 'react';
import Tippy from '@tippy.js/react';

import { Palette } from 'Components/Palette';

import { Container, Circle } from './styled';

const ColorPicker: React.FC<{
  activeColor: string;
  onChange: (color: string) => void;
}> = ({ activeColor, onChange }) => {
  const [color, setColor] = useState(activeColor);
  const [isVisible, setIsVisible] = useState(false);

  useEffect(() => {
    if (color === activeColor) {
      return;
    }

    onChange(color);
  }, [color]);

  return (
    <Tippy
      interactive={true}
      placement={'bottom-end'}
      trigger={'click'}
      hideOnClick={true}
      onShow={() => setIsVisible(true)}
      onHide={() => setIsVisible(false)}
      content={<Palette activeColor={color} setColor={setColor} />}
    >
      <Container isActive={isVisible}>
        <Circle style={{ background: color }} />
      </Container>
    </Tippy>
  );
};

export { ColorPicker };
