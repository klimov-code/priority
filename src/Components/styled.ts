import styled from 'styled-components';

const Text = styled.span`
  display: block;
  font-size: 13px;
  line-height: 18px;
  color: #454c54;
`;

const TextBold = styled(Text)`
  font-weight: bold;
`;

export { Text, TextBold };
