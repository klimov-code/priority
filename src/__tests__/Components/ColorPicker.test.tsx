import React from 'react';
import { render } from 'react-testing-library';
import { ColorPicker } from 'Components/ColorPicker';

describe('ColorPicker', () => {
  it('renders correctly', () => {
    const onChange = jest.fn();
    const activeColor = '#4a4a4a';

    const { container } = render(<ColorPicker onChange={onChange} activeColor={activeColor} />);

    expect(container).toMatchSnapshot();
  });
});
