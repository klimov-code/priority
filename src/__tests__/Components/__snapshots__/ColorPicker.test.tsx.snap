// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker renders correctly 1`] = `
<div>
  <div
    class="sc-ifAKCX hOJEll"
    tabindex="0"
  >
    <div
      class="sc-EHOje cpQrJT"
      style="background: rgb(74, 74, 74);"
    />
  </div>
</div>
`;
